features/246-custom-domain/index.md

```diff
@@ -0,0 +1,70 @@
+# Custom domain
+
+## Overview
+
+Allow developer to setup custom domain for their application, basically following features will be supported.
+
+- Allow developer to setup custom domain.
+- Let's Encrypt certificate will be provided by default, certificate will be renew automatically.
+- Developer can configure to use custom SSL certificates for their domain. They will need to update the certificates before it expires in this case.
+
+## Use Cases
+
+### Setup custom domain through skycli
+
+1. Add custom domain
+
+    ```sh
+    $ skycli domain add myapp.example.com --app=myapp
+    Added domain myapp.example.com successfully!
+
+    Add following DNS records in your DNS provider.
+
+    TYPE      HOST                     VALUE
+    TXT       _skygear.example.com     5636486ffc5a4dfebf4a13f480bd9a95
+    A         myapp.example.com          <ingress controller lb ip>
```

Comment:
It seems many providers use CNAME, is there any reason we use A & TXT instead?

Reply:
Since we will want to support both apex domain and subdomains. I have thought of using ANAME or ALIAS for apex domain, but not all providers support this. At least route 53 doesn't support this.